client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,162 @@
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const handleToggleForSimpleSite = ( checked: boolean ) => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		simpleSiteTogglePostByEmail( checked, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Settings saved successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem saving your changes.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		if ( siteIsJetpack ) {
+			dispatch( jetpackRegeneratePostByEmail( selectedSiteId ) );
+			return;
+		}
+
+		simpleSiteRegeneratePostByEmail( undefined, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Address regenerated successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem regenerating the address.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	let email;
+	if ( siteIsJetpack ) {
+		email = address && address !== 'regenerate' ? address : '';
+	} else {
+		email = simpleSitePostByEmailSettings?.email;
+	}
+	const isActive = siteIsJetpack
+		? jetpackPostByEmailIsActive
+		: simpleSitePostByEmailSettings?.isEnabled;
+	const isJetpackDisabled = isFormPending || moduleUnavailable;
+	const isSimpleSiteDisabled =
+		isFormPending || isSimpleSitePendingToggle || isSimpleSitePendingRegenerate;
+	const isDisabledControls = ! isActive || isJetpackDisabled || isSimpleSiteDisabled;
+	const toggleLabel = translate( 'Publish posts by sending an email' );
+
+	return (
+		<>
+			<FormFieldset>
+				<SupportInfo
+					text={ translate(
+						'Allows you to publish new posts by sending an email to a special address.'
+					) }
+					link={
+						siteIsAtomic
+							? localizeUrl( 'https://wordpress.com/support/post-by-email/' )
+							: 'https://jetpack.com/support/post-by-email/'
+					}
+					privacyLink={ ! siteIsAtomic }
+				/>
+				{ siteIsJetpack ? (
+					<JetpackModuleToggle
+						siteId={ selectedSiteId }
+						moduleSlug="post-by-email"
+						label={ toggleLabel }
+						disabled={ isJetpackDisabled }
+					/>
+				) : (
+					<ToggleControl
+						checked={ !! simpleSitePostByEmailSettings?.isEnabled }
+						disabled={ isSimpleSiteDisabled }
+						label={ toggleLabel }
+						onChange={ handleToggleForSimpleSite }
+					/>
+				) }
+
+				<div className="publishing-tools__module-settings site-settings__child-settings">
+					<FormLabel
+						className={ clsx( {
+							'is-disabled': isDisabledControls,
+						} ) }
+					>
+						{ translate( 'Send your new posts to this email address:' ) }
+					</FormLabel>
+					<ClipboardButtonInput
+						className="publishing-tools__email-address"
+						disabled={ isDisabledControls }
+						value={ email }
+					/>
+					<Button onClick={ handleRegenerate } disabled={ isDisabledControls }>
+						Regenerate address
```

Comment:
There was previously (including on `trunk`) logic to change this button content to `Regenerating...` while a new address was being generated for Jetpack sites. I've removed that in favor of consistent button text for the reasons below.

It isn't working for Jetpack (including on trunk) as Fredrik and I noted above, and when I applied it to Simple sites (where it _does_ work) I really didn't like the way it changed the size of the button. The visual jump was king of jarring. Could be fixed with CSS, but it honestly didn't feel like it was worth it.

On Simple, we're also now temporarily disabling the button while the new address is generated, so there's visual feedback in that form, and if/when the `isRegeneratingJetpackPostByEmail` is working in the future, we can apply the temp disable effect there, too.

tl;dr: from a user perspective, removing this text change doesn't impact Jetpack users at all, and we have something else in place for Simple that I think works better anyway.